src/test/java/redis/clients/jedis/util/JedisURIHelperTest.java

```diff
@@ -73,4 +72,13 @@ public void shouldGetProtocolFromDefinition() {
     assertEquals(RedisProtocol.RESP3, getRedisProtocol(URI.create("redis://host:1234/1?protocol=3")));
     assertEquals(RedisProtocol.RESP3, getRedisProtocol(URI.create("redis://host:1234/1/?protocol=3")));
   }
+
+  @Test
+  public void shouldReturnNullIfURIDoesNotHavePassword() throws URISyntaxException {
+    URI uri = new URI("redis://user@host:9000/0");
+    JedisException jedisException = assertThrows(JedisException.class, () -> {
+      getPassword(uri);
+    });
+    assertEquals(jedisException.getMessage(), "AUTH error. Password not provided in uri");
+  }
```

Comment:
```suggestion
  @Test
  public void shouldReturnEmptyPassword() {
    // ensure we can provide an empty password for default user
    assertThat( JedisURIHelper.getPassword(URI.create("redis://:@host:9000/0")),  emptyString());
  }

  @Test
  public void shouldThrowIfNoPasswordInURI() throws URISyntaxException {
    // ensure we throw if user is provided but password is missing in URI
    URI uri = new URI("redis://user@host:9000/0");
    IllegalArgumentException jedisException = assertThrows(IllegalArgumentException.class, () -> {
      getPassword(uri);
    });
    assertEquals(jedisException.getMessage(), "Password not provided in the URI");
  }
```
Update the test to use IllegalArgumentException

Reply:
Sounds good. Let me push review suggestions